### PR TITLE
[PM-6018] Error message when adding or removing an existing client organization

### PR DESCRIPTION
**Pull request by cyprain-okeke**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-6018
## 📔 Objective

Resolve the issue of not able to add or remove an existing client organization presumably only if there is no subscription attached to the organization. One example of this behavior would be if the organization was using a manual trial.

## 📸 Screenshots

https://github.com/bitwarden/server/assets/108260115/de47dd44-65f0-4908-bca5-f8ae8c05f61a

https://github.com/bitwarden/server/assets/108260115/14d9dd03-79a4-43ba-b975-cfdc381acbfc

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

https://github.com/bitwarden/server/assets/108260115/d2eb6b80-eec0-4c6e-833e-dac137512f7a

### CODE REVIEW

**Review by eliykat — APPROVED** [review]
This seems OK to me, however this is effectively billing code, so please wait for a review from Billing Team as well.

More generally, I'd like a better encapsulation of billing code in Billing Team owned services. There's a lot of billing code in `ProviderService` that AC Team don't know much about, at a glance a lot of it could be moved to `OrganizationBillingService` or similar. I would like our level of knowledge (in AC Team) to be "this impacts the subscription, so I'll call `OrganizationBillingService.UpdateSubscription`", not "first I have to apply the provider price rate, then check if they have a gateway id, and if they do then I'll call the stripe adapter..." etc.

**Review by amorask-bitwarden — APPROVED** [review]

**Pull request merged**
